src/main/java/org/jvnet/hudson/test/InboundAgentRule.java

```diff
@@ -447,124 +447,74 @@ public AgentArguments(@NonNull String agentJnlpUrl, @NonNull File agentJar, @Non
     }
 
-    private static class WaitForAgentOnline implements RealJenkinsRule.Step {
-        private final String name;
-        private final Map<String, Level> loggers;
-
-        WaitForAgentOnline(String name, Map<String, Level> loggers) {
-            this.name = name;
-            this.loggers = loggers;
+    private static void waitForAgentOnline(JenkinsRule r, String name, Map<String, Level> loggers) throws Exception {
+        Node node = r.jenkins.getNode(name);
+        if (node == null) {
+            throw new AssertionError("no such agent: " + name);
         }
-
-        @Override
-        public void run(JenkinsRule r) throws Throwable {
-            wait(r, name, loggers);
+        if (!(node instanceof Slave)) {
+            throw new AssertionError("agent is not a Slave: " + name);
         }
-
-        static void wait(JenkinsRule r, String name, Map<String, Level> loggers) throws Exception {
-            Node node = r.jenkins.getNode(name);
-            if (node == null) {
-                throw new AssertionError("no such agent: " + name);
-            }
-            if (!(node instanceof Slave)) {
-                throw new AssertionError("agent is not a Slave: " + name);
-            }
-            r.waitOnline((Slave) node);
-            if (!loggers.isEmpty()) {
-                VirtualChannel channel = node.getChannel();
-                assert channel != null;
-                channel.call(new JenkinsRule.RemoteLogDumper(null, loggers, false));
-            }
+        r.waitOnline((Slave) node);
+        if (!loggers.isEmpty()) {
+            VirtualChannel channel = node.getChannel();
+            assert channel != null;
+            channel.call(new JenkinsRule.RemoteLogDumper(null, loggers, false));
         }
     }
 
-    private static class WaitForAgentOffline implements RealJenkinsRule.Step {
-        private final String name;
-        WaitForAgentOffline(String name) {
-            this.name = name;
-        }
-
-        @Override
-        public void run(JenkinsRule r) throws Throwable {
-            stop(r, name);
-        }
-
-        private static void stop(JenkinsRule r, String name) throws InterruptedException {
-            Computer c = r.jenkins.getComputer(name);
-            if (c != null) {
-                while (c.isOnline()) {
-                    Thread.sleep(100);
-                }
+    private static void waitForAgentOffline(JenkinsRule r, String name) throws InterruptedException {
+        Computer c = r.jenkins.getComputer(name);
+        if (c != null) {
+            while (c.isOnline()) {
+                Thread.sleep(100);
             }
         }
     }
 
-    private static class CreateAgent implements RealJenkinsRule.Step2<String> {
-        private final Options options;
-
-        CreateAgent(Options options) {
-            this.options = options;
-        }
+    private static String createAgentRJR(JenkinsRule r, Options options) throws Throwable {
```

Comment:
not to conflict with https://github.com/jenkinsci/jenkins-test-harness/blob/7a4133525c7591d025da037fd2ed3c19a34c9e8c/src/main/java/org/jvnet/hudson/test/InboundAgentRule.java#L269